account_invoice_supplier_self_invoice/models/res_partner.py

```diff
@@ -115,6 +118,4 @@ def action_set_self_invoice(self):
     @api.onchange("self_invoice")
     def onchange_self_invoice(self):
         if self.self_invoice and not self.self_invoice_report_footer:
-            self.self_invoice_report_footer = self._fields[
-                "self_invoice_report_footer"
-            ].args["default"]
+            self.self_invoice_report_footer = self._default_self_invoice_report_footer()
```

Comment:
I think it's better to use the partner language instead of user language to generate the translated value.

```suggestion
            self.self_invoice_report_footer = self.with_context(lang=self.lang)._default_self_invoice_report_footer()
```

`_` translation method uses lang in context

Reply:
Hello @etobella , i think this is a good point. The language should be from partner, not the user. I guess you agree too. Let us know, thanks!

Reply:
:+1: